fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -62,38 +62,41 @@ impl<T> Interpreter<PredicateStorage, T> {
 
         vm.init_predicate(checked.clone());
 
-        // Needed for now because checked is only freed once the value is collected into a Vec
-        #[allow(clippy::needless_collect)]
-        let predicates: Vec<_> = (0..checked.transaction().inputs().len())
-            .filter_map(|i| RuntimePredicate::from_tx(&params, checked.transaction(), i))
-            .collect();
+        for (idx, input) in checked.transaction().inputs().iter().enumerate() {
+            if let Some(predicate) = RuntimePredicate::from_tx(&params, checked.transaction(), idx) {
+                // VM is cloned because the state should be reset for every predicate verification
+                let mut vm = vm.clone();
 
-        for predicate in predicates {
-            // VM is cloned because the state should be reset for every predicate verification
-            let mut vm = vm.clone();
+                let gas_used = input.predicate_gas_used().unwrap();
```

Comment:
Is this unwrap supposed to be here?

Reply:
yes, this field is not present on all inputs so could be None

Reply:
But this is production code, we don't want to panic the whole client because of this right?

Reply:
fixed